### ISSUE TITLE
ENH Add preprocessing algorithm "Certifying and removing disparate impact" by Feldman et al.

### ISSUE THREAD

**Issue by hildeweerts** [fairlearn/fairlearn#1022]
The paper [Certifying and removing disparate impact](https://arxiv.org/abs/1412.3756) by Feldman et al. introduces a simple pre-processing procedure (section 5 of the paper). The goal of this task is to implement this procedure to Fairlearn. 

The main intuition of this algorithm is to map the distribution of Y of each sensitive group to the median distribution of Y. In this way, the new percentile rank of each instance in the data set will be the same as the percentile rank of that instance within their sensitive group. See also this example from the paper, where Y refers to SAT scores:

<img width="785" alt="image" src="https://user-images.githubusercontent.com/24417440/154045702-7134d694-84f3-4e87-b753-2ef7cdca6f15.png">

The algorithm is also part of AIF360 where it is known as [`DisparateImpactRemover`](https://aif360.readthedocs.io/en/latest/modules/generated/aif360.algorithms.preprocessing.DisparateImpactRemover.html). To avoid confusion with legal use of the term 'disparate impact' I would propose to use a different name, e.g., something related to the mapping of percentiles?

Completing this item requires:
- [ ] preprocessing technique code in `fairlearn.preprocessing`
- [ ] unit tests in `test.unit.preprocessing`
- [ ] descriptive API reference (directly in the docstring)
- [ ] a short user guide in `docs.user_guide.mitigation.rst`

A fully fledged example notebook is not required.

To claim this task please respond below. Of course, you can also ask questions!

**Comment by quinnarnold** [comment]
I am interested in working on this, but because the issue predates the current preprocessing APIs I would like to confirm the intended interface before starting.

For a minimal first version, I propose the ordered/numeric repair from Section 5 and geometric partial repair from Section 5.1.2, initially for one sensitive feature. `repair_level=0` would leave values unchanged and `repair_level=1` would apply full repair. Unordered categorical repair and multiple sensitive features would remain out of scope unless required.

The paper and AIF360 expose repair as a dataset-level operation, while Fairlearn preprocessing generally follows scikit-learn transformer conventions. Would you prefer:

1. A dataset-level `fit_transform(X, sensitive_features=...)` operation faithful to that presentation, or
2. A fitted transformer with sensitive columns included in `X` and identified in the constructor, similar to `CorrelationRemover`?

Would `QuantileRepairer` be an acceptable name? Once the API is chosen, I would keep the first PR to the numeric/ordered, one-sensitive-feature, geometric-repair scope and add deterministic tests for exact mappings, rank preservation, partial repair, ties, and unequal group sizes.
